core/nodejsActionBase/runner.js

```diff
@@ -19,63 +19,91 @@
  * an action.
  *
  * This file (runner.js) must currently live in root directory for nodeJsAction.
- * See issue #102 for a discussion.
  */
-var util  = require('util');
+var util = require('util');
 
-function NodeActionRunner(message, whisk) {
-    this.userScriptName = getname(message.name);
+function NodeActionRunner(whisk) {
+    // Use this ref inside lambdas etc.
+    var thisRunner = this;
+
+    this.userScriptName = undefined;
     this.userScriptMain = undefined;
-    this.initError = undefined;
 
-    try {
-        eval(message.code);
-        this.userScriptMain = eval(message.main);
-    } catch (e) {
-        this.initError = e;
-        return;
-    }
+    // This structure is reset for every action invocation. It contains two fields:
+    //   - completed; indicating whether the action has already signaled completion
+    //   - next; a callback to be invoked with the result of the action.
+    // Note that { error: ... } results are still results.
+    var callback = {
+        completed : undefined,
+        next      : function (result) { return; }
+    };
 
-    var callback = { };
+    this.whisk = modWhisk(whisk, callback);
 
-    function closeAndReturn(result) {
+    this.init = function(message) {
+        // Determining a sensible name for the action.
+        var name = typeof message.name === 'string' ? message.name.trim() : '';
+        name = name !== '' ? name : 'Anonymous User Action';
 
-        if (typeof callback.closedConnection === 'undefined') {
-            callback.closedConnection = true;
-            callback.next(result);
-        } else {
-            // There is no 'else': we can't close the connection more than once
-            // (which is what callback.next eventually does). We should warn the
-            // user if (s)he is trying to do just that, but that warning takes
-            // place in the call to (overwritten) _terminate.
-        }
+        this.userScriptName = name;
+
+        // Loading the user code.
+        return new Promise(
+            function (resolve, reject) {
+                try {
+                    eval(message.code);
+                    thisRunner.userScriptMain = eval(message.main);
+
+                    if (typeof thisRunner.userScriptMain === 'function') {
+                        resolve(true);
```

Comment:
Resolving on `true` is not meaningful to the caller (since this is the only value that is ever resolved from this function) and can be confusing to the reader who may be looking for what circumstances `false` is ever used.